components/sidebar.py

```diff
@@ -0,0 +1,139 @@
+"""
+Copyright 2018-2020 Faculty Science Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Code modified from https://github.com/facultyai/dash-bootstrap-components
+"""
+import dash_bootstrap_components as dbc
+import dash_html_components as html
+from dash.dependencies import Input, Output, State
+
+from app import app
+
+
+page_names = ['Upload data', 'Map', 'Scatter plot']
+page_paths = ['/upload', '/map', '/scatter']
+
+
+def get_component():
+    """Return the sidebar component."""
+    return html.Div(
+        [
+            get_sidebar_header(),
+            # use the Collapse component to animate hiding / revealing links
+            dbc.Collapse(
+                dbc.Nav(
+                    [
+                        dbc.NavLink(
+                            page_names[0],
+                            href=page_paths[0],
+                            id="page-0-link"),
+
+                        dbc.NavLink(
+                            page_names[1],
+                            href=page_paths[1],
+                            id="page-1-link"),
+
+                        dbc.NavLink(
+                            page_names[2],
+                            href=page_paths[2],
+                            id="page-2-link"),
+                    ],
```

Comment:
I'm not sure about whether this would be better implemented with list comprehension. In its current form, it's still easy to read, but if we end up with lots and lots of items in the menu, it will make it harder to look at the whole picture